<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QtWidgetsApplication1Class</class>
 <widget class="QMainWindow" name="QtWidgetsApplication1Class">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1126</width>
    <height>931</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>QtWidgetsApplication1</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>1</horstretch>
     <verstretch>1</verstretch>
    </sizepolicy>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::DefaultContextMenu</enum>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="OpenGLCustomWidget" name="openGLWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>400</width>
         <height>400</height>
        </size>
       </property>
       <property name="acceptDrops">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QWidget" name="">
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>20</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="editCameraPositionLayout">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item>
           <widget class="QLabel" name="label_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <family>Roboto Black</family>
              <pointsize>14</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="text">
             <string>CameraPosition</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QDoubleSpinBox" name="cameraXSpin">
              <property name="accelerated">
               <bool>true</bool>
              </property>
              <property name="showGroupSeparator" stdset="0">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-9999999999999999538762658202121142272.000000000000000</double>
              </property>
              <property name="maximum">
               <double>100000000000000000620008645040778319495168.000000000000000</double>
              </property>
              <property name="stepType">
               <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="cameraYSpin">
              <property name="accelerated">
               <bool>true</bool>
              </property>
              <property name="showGroupSeparator" stdset="0">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-9999999999999999538762658202121142272.000000000000000</double>
              </property>
              <property name="maximum">
               <double>100000000000000000620008645040778319495168.000000000000000</double>
              </property>
              <property name="stepType">
               <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="cameraZSpin">
              <property name="accelerated">
               <bool>true</bool>
              </property>
              <property name="showGroupSeparator" stdset="0">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-9999999999999999538762658202121142272.000000000000000</double>
              </property>
              <property name="maximum">
               <double>100000000000000000620008645040778319495168.000000000000000</double>
              </property>
              <property name="stepType">
               <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="addVertexLayout">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item>
           <widget class="QLabel" name="label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <family>Roboto Black</family>
              <pointsize>14</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="text">
             <string>Vertex</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QDoubleSpinBox" name="vertexXSpin">
              <property name="accelerated">
               <bool>true</bool>
              </property>
              <property name="showGroupSeparator" stdset="0">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-9999999999999999538762658202121142272.000000000000000</double>
              </property>
              <property name="maximum">
               <double>100000000000000000620008645040778319495168.000000000000000</double>
              </property>
              <property name="stepType">
               <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="vertexYSpin">
              <property name="accelerated">
               <bool>true</bool>
              </property>
              <property name="showGroupSeparator" stdset="0">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-9999999999999999538762658202121142272.000000000000000</double>
              </property>
              <property name="maximum">
               <double>100000000000000000620008645040778319495168.000000000000000</double>
              </property>
              <property name="stepType">
               <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="vertexZSpin">
              <property name="accelerated">
               <bool>true</bool>
              </property>
              <property name="showGroupSeparator" stdset="0">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-9999999999999999538762658202121142272.000000000000000</double>
              </property>
              <property name="maximum">
               <double>100000000000000000620008645040778319495168.000000000000000</double>
              </property>
              <property name="stepType">
               <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPushButton" name="addVertexButton">
            <property name="text">
             <string>Add vertex</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OpenGLCustomWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">openglcustomwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="QtWidgetsApplication1.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>addVertexButton</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>onAddVertexClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1114</x>
     <y>163</y>
    </hint>
    <hint type="destinationlabel">
     <x>435</x>
     <y>500</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cameraXSpin</sender>
   <signal>valueChanged(double)</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>onCameraPositionSpinChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>848</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>562</x>
     <y>465</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cameraYSpin</sender>
   <signal>valueChanged(double)</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>onCameraPositionSpinChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>955</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>562</x>
     <y>465</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cameraZSpin</sender>
   <signal>valueChanged(double)</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>onCameraPositionSpinChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1063</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>562</x>
     <y>465</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onAddVertexClicked()</slot>
  <slot>onCameraPositionSpinChanged()</slot>
 </slots>
</ui>
